### PR TITLE
feat: add touchstart, touchmove and touchend methods

### PR DESCRIPTION
**Pull request by charlieinitialdigital**
**What kind of change does this PR introduce?**

add touchstart,touchmove,touchend 

**Did you add tests for your changes?**

yes

**If relevant, did you update the documentation?**

yes

**Summary**

for now, you can only tap in the phone, but when it comes to some scrolling stuff, you can't test by puppeteer

**Does this PR introduce a breaking change?**

no

**Other information**

### CODE REVIEW

File: test/src/touchscreen.spec.ts [hunk 1]
```diff
@@ -26,29 +26,41 @@ describe('Touchscreen', function () {
...
-  it('should tap the button', async () => {
+  it('should report touches', async () => {
     const {page, server} = getTestState();
     const iPhone = KnownDevices['iPhone 6']!;
     await page.emulate(iPhone);
-    await page.goto(server.PREFIX + '/input/button.html');
-    await page.tap('button');
+    await page.goto(server.PREFIX + '/input/touches.html');
+    const button = (await page.$('button'))!;
+    await button.tap();
```
Comment (OrKoN) [inline test/src/touchscreen.spec.ts]: let's not change the existing test. Please add a new test for the ElementHandle.tap

File: packages/puppeteer-core/src/common/Input.ts [hunk 2]
```diff
@@ -682,4 +682,42 @@ export class Touchscreen {
       modifiers: this.#keyboard._modifiers,
     });
   }
+
+  /**
+   * Dispatches a `touchstart` event.
+   * @param x - Horizontal position of the tap.
+   * @param y - Vertical position of the tap.
+   */
+  async touchStart(x: number, y: number): Promise<void> {
```
Comment (OrKoN) [inline packages/puppeteer-core/src/common/Input.ts:682]: let's use touchStart and touchEnd in tap above

File: packages/puppeteer-core/src/common/Input.ts [hunk 3]
```diff
@@ -682,4 +682,42 @@ export class Touchscreen {
...
+    const movePoints = [{x: Math.round(x), y: Math.round(y)}];
+    await this.#client.send('Input.dispatchTouchEvent', {
+      type: 'touchMove',
+      touchPoints: movePoints,
+      modifiers: this.#keyboard._modifiers,
+    });
+  }
+  /**
+   * Dispatches a `touchend` event.
+   * stop the touchStart function
```
Comment (OrKoN) [inline packages/puppeteer-core/src/common/Input.ts]: Let's remove this part of the comment and leave only `Dispatches a `touchend` event.`

Comment (charlieinitialdigital) [reply packages/puppeteer-core/src/common/Input.ts]: ok

File: packages/puppeteer-core/src/common/Input.ts (hunk 2 above)
Comment (charlieinitialdigital) [reply packages/puppeteer-core/src/common/Input.ts:682]: ok

File: packages/puppeteer-core/src/common/Input.ts [hunk 4]
```diff
@@ -670,17 +670,8 @@ export class Touchscreen {
...
-      type: 'touchStart',
-      touchPoints,
-      modifiers: this.#keyboard._modifiers,
-    });
-    await this.#client.send('Input.dispatchTouchEvent', {
-      type: 'touchEnd',
-      touchPoints: [],
-      modifiers: this.#keyboard._modifiers,
-    });
+    this.touchStart(x, y);
```
Comment (OrKoN) [inline packages/puppeteer-core/src/common/Input.ts]: the result of the calls needs to be `await`ed

File: test/src/touchscreen.spec.ts (hunk 1 above)
Comment (charlieinitialdigital) [reply test/src/touchscreen.spec.ts]: ok, I reAdd the previous one

File: packages/puppeteer-core/src/common/Input.ts (hunk 4 above)
Comment (charlieinitialdigital) [reply packages/puppeteer-core/src/common/Input.ts]: sorry for my careless

**Review by OrKoN — APPROVED** [review]

**Comment by OrKoN** [comment]
Looks like "Touchscreen should report touchMove" is failing on Firefox. Could you check that? if it's not supported in Firefox, could you please add the failing test to TestExpectations.json in the test folder.

**Comment by charlieinitialdigital** [comment]
sure

**Comment by OrKoN** [comment]
Thanks for the PR!

**Pull request merged**